From a69d400b15b449da8de16bbe08eb44c81d741fa9 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Fri, 12 Sep 2014 18:07:33 -0400
Subject: [powerpc] Increase EEH recovery timeout for SR-IOV

Message-id: <1410545655-205645-225-git-send-email-dzickus@redhat.com>
Patchwork-id: 94418
O-Subject: [RHEL7 PATCH 224/626] powerpc: Increase EEH recovery timeout for SR-IOV
Bugzilla: 1127366
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: David Gibson <dgibson@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1127366

commit fb48dc22824daaa60ff1d6a6c9e22c79112dfb8e
Author: Brian King <brking@linux.vnet.ibm.com>
Date:   Mon Nov 25 16:27:54 2013 -0600

    powerpc: Increase EEH recovery timeout for SR-IOV

    In order to support concurrent adapter firmware download
    to SR-IOV adapters on pSeries, each VF will see an EEH event
    where the slot will remain in the unavailable state for
    the duration of the adapter firmware update, which can take
    as long as 5 minutes. Extend the EEH recovery timeout to
    account for this.

    Signed-off-by: Brian King <brking@linux.vnet.ibm.com>
    Acked-by: Gavin Shan <shangw@linux.vnet.ibm.com>
    Signed-off-by: Benjamin Herrenschmidt <benh@kernel.crashing.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/powerpc/kernel/eeh.c b/arch/powerpc/kernel/eeh.c
index 52e1520..dda06a6 100644
--- a/arch/powerpc/kernel/eeh.c
+++ b/arch/powerpc/kernel/eeh.c
@@ -84,7 +84,7 @@
 #define EEH_MAX_FAILS 2100000
 
 /* Time to wait for a PCI slot to report status, in milliseconds */
-#define PCI_BUS_RESET_WAIT_MSEC (60*1000)
+#define PCI_BUS_RESET_WAIT_MSEC (5*60*1000)
 
 /* Platform dependent EEH operations */
 struct eeh_ops *eeh_ops = NULL;
diff --git a/arch/powerpc/kernel/eeh_driver.c b/arch/powerpc/kernel/eeh_driver.c
index 0a2586c..72e9046 100644
--- a/arch/powerpc/kernel/eeh_driver.c
+++ b/arch/powerpc/kernel/eeh_driver.c
@@ -480,7 +480,7 @@ static int eeh_reset_device(struct eeh_pe *pe, struct pci_bus *bus)
 /* The longest amount of time to wait for a pci device
  * to come back on line, in seconds.
  */
-#define MAX_WAIT_FOR_RECOVERY 150
+#define MAX_WAIT_FOR_RECOVERY 300
 
 static void eeh_handle_normal_event(struct eeh_pe *pe)
 {
-- 
1.7.1